aaelf64/aaelf64.rst

```diff
@@ -1001,20 +1001,25 @@ See also table `GOT-relative data relocations`_.
   +---------+---------+----------------------+-----------+------------------------------------------------------------------+
-  | 258     | 1       | R\_<CLS>\_ABS32      | S + A     | Check that -2\ :sup:`31` <= X < 2\ :sup:`32`                     |
+  | 258     | 1       | R\_<CLS>\_ABS32      | S + A     | Write bits [0:31] of X at byte alignment.                        |
+  |         |         |                      |           | Check that -2\ :sup:`31` <= X < 2\ :sup:`32`.                    |
   +---------+---------+----------------------+-----------+------------------------------------------------------------------+
-  | 259     | 2       | R\_<CLS>\_ABS16      | S + A     | Check that -2\ :sup:`15` <= X < 2\ :sup:`16`                     |
+  | 259     | 2       | R\_<CLS>\_ABS16      | S + A     | Write bits [0:15] of X at byte alignment.                        |
+  |         |         |                      |           | Check that -2\ :sup:`15` <= X < 2\ :sup:`16`.                    |
   +---------+---------+----------------------+-----------+------------------------------------------------------------------+
-  | 260     | \-      | R\_<CLS>\_PREL64     | S + A - P | No overflow check                                                |
+  | 260     | \-      | R\_<CLS>\_PREL64     | S + A - P | Write bits [0:63] of X at byte alignment.  No overflow check.    |
   +---------+---------+----------------------+-----------+------------------------------------------------------------------+
-  | 261     | 3       | R\_<CLS>\_PREL32     | S + A - P | Check that -2\ :sup:`31` <= X < 2\ :sup:`32`                     |
+  | 261     | 3       | R\_<CLS>\_PREL32     | S + A - P | Write bits [0:31] of X at byte alignment.                        |
+  |         |         |                      |           | Check that -2\ :sup:`31` <= X < 2\ :sup:`32`.                    |
   +---------+---------+----------------------+-----------+------------------------------------------------------------------+
-  | 262     | 4       | R\_<CLS>\_PREL16     | S + A - P | Check that -2\ :sup:`15` <= X < 2\ :sup:`16`                     |
+  | 262     | 4       | R\_<CLS>\_PREL16     | S + A - P | Write bits [0:16] of X at byte alignment.                        |
+  |         |         |                      |           | Check that -2\ :sup:`15` <= X < 2\ :sup:`16`.                    |
   +---------+---------+----------------------+-----------+------------------------------------------------------------------+
-  | 314     | 29      | R\_<CLS>\_PLT32      | S + A - P | Check that -2\ :sup:`31` <= X < 2\ :sup:`31` see call and jump   |
-  |         |         |                      |           | relocations                                                      |
+  | 314     | 29      | R\_<CLS>\_PLT32      | S + A - P | Write bits [0:31] of X at byte alignment.                        |
```

Comment:
In this case the PLT32 is designed for relative vtable entries which are always 4-byte aligned. If we enforced that then we could enforce that the result was also 4-byte aligned. However that would tie the relocation to its current use case.

Reply:
Thanks -- I didn't have a great handle on where this is usually used.
Based on what you said I personally would prefer to not add that restriction based on leaving flexibility for other use-cases.
Do you have a preference?
(I read your comment as mentioning possibilities rather than suggesting a change -- so didn't notice a preference in it).

FWIW the current implementation in `lld` (`ld.bfd` does not seem to implement this) allows 1-byte aligned values.

Reply:
I'm happy to leave as is. AFAIK the only production use of PLT32 is in Fuchsia for relative vtables so probably not a high priority for ld.bfd.